New integraton iNELS

jhoralek opened the pull request:
- create test area
  - create component
  - basic config flow based on manual configuration
  - translation for config flow
  - set up manifest, with requirements

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request:  https://github.com/home-assistant/home-assistant.io/pull/23624

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [x] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

jhoralek committed:
- scaffolding project
  - create test area
  - create component
  - basic config flow based on manual configuration
  - translation for config flow
  - set up manifest, with requirements

homeassistant commented:
Hi @jhoralek, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#75091).

Once you do that we will be able to review and accept this pull request.

Thanks!

jhoralek committed:
## Inels integration
@jhoralek

- new integration in HA with inels system over mqtt protocol

- config_flow with mqtt broker connection requirements
- config_flow test
- Inels coordinator to handle devices
- Switch platform

jhoralek committed:
## Inels integration
@jhoralek

- new integration in HA with inels system over mqtt protocol

- config_flow with mqtt broker connection requirements
- config_flow test
- Inels coordinator to handle devices
- Switch platform

jhoralek committed:
- fixing
  - switch initializing
  - data binding
  - icon

- new
  - sensor primary temp sensor based on the type
  - data parsing from hex to decimal

- config flow
  - add websocket protocol

- remove icon from base class

bachya commented:
Rebase the PR off of the latest `dev` to remove the unrelated changes.

jhoralek committed:
Merge branch 'dev' into inels/jh/init_integration

jhoralek committed:
Merge branch 'home-assistant:dev' into inels/jh/init_integration

jhoralek committed:
Merge branch 'dev' into inels/jh/init_integration

jhoralek committed:
- light platform with dimmable support

jhoralek committed:
Merge branch 'inels/jh/init_integration' of https://github.com/Nejezchleb/core into inels/jh/init_integration

jhoralek commented:
> Rebase the PR off of the latest `dev` to remove the unrelated changes.

Done. Everything seems to be all right.

jhoralek committed:
- new platform cover
- bump version of inels-mqtt lib

jhoralek committed:
Merge branch 'home-assistant:dev' into inels/jh/init_integration

jhoralek committed:
Merge branch 'inels/jh/init_integration' of https://github.com/Nejezchleb/core into inels/jh/init_integration

jhoralek committed:
Merge branch 'home-assistant:dev' into inels/jh/init_integration

jhoralek committed:
Merge branch 'home-assistant:dev' into inels/jh/init_integration

jhoralek committed:
Merge branch 'home-assistant:dev' into inels/jh/init_integration

jhoralek committed:
Merge branch 'home-assistant:dev' into inels/jh/init_integration

jhoralek committed:
Merge branch 'home-assistant:dev' into inels/jh/init_integration

MartinHjelmare commented:
Please open a new PR on a freshly synced branch. Currently all github users that were requested for review are getting notifications every time this PR is updated, spamming us. Review hasn't started yet, so there is no history to keep intact yet.

I'll close this PR now.
